### ISSUE TITLE
Database.UnlockProtectedEntries() panics if password is invalid (KeeWeb)

### ISSUE THREAD

**Issue by dontpullthis** [tobischo/gokeepasslib#90]
Hello,

I am trying to read a `*.kdbx` file following tutorial at this page:
https://github.com/tobischo/gokeepasslib/blob/master/examples/reading/example.go#L17
If password is correct, then the database file opens without any problems, but in case of invalid password the application throws panic.

Debugger indicates that the panic is thrown at this line:
https://github.com/tobischo/gokeepasslib/blob/master/database.go#L101
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x55fc21]
```
According to `variables` section in debugger, `db.Content.InnerHeader` equals `nil` at this line, therefore the app cannot access to `db.Content.InnerHeader.InnerRandomStreamID`.

Is it possible to handle such cases in `Database.GetStreamManager` function? I guess it might be something like
```go
if db.Header.IsKdbx4() {
    if db.Content.InnerHeader == nil {
       return nil, errors.New("Cannot read the header data. Either credentials are invalid or the database file is corrupted.")
    }

    return NewStreamManager(
        db.Content.InnerHeader.InnerRandomStreamID,
        db.Content.InnerHeader.InnerRandomStreamKey,
    )
}
```

The file I am trying to read is created by KeeWeb app: https://keeweb.info . Sharing it here for debugging purposes:
https://drive.google.com/file/d/1gUrYuKa-lfK-cLiv3Ambo7UHZV1tOV7t/view?usp=sharing

**Comment by tobischo** [comment]
Yeah, it should be handleable.

It is quite interesting that it even gets that far with invalid credentials.

Did you try and see what happens when you open it with the original Keepass2 and provide the same inputs?

**Comment by tobischo** [comment]
addressed in 01e10cdd02e7e6987dc7ffd6ef1c24c453c05f52

**Comment by dontpullthis** [comment]
Unfortunately, I can't test it in Keepass 2, because I have Keeweb only.

Anyway, thanks for fix, the app doesn't crash anymore after pulling the recent changes from master branch!
